src/main/java/apoc/diff/Diff.java

```diff
@@ -0,0 +1,110 @@
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.procedure.*;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * @author Benjamin Clauss
+ * @since 29.03.2018
+ */
+public class Diff {
+
+    @Context
+    public GraphDatabaseService db;
+
+    @UserFunction
+    @Description("apoc.diff.maps([leftMap],[rightMap])" )
+    public boolean maps(@Name("leftMap") Map<String, Object> leftMap, @Name("rightMap") Map<String, Object> rightMap) {
+        return !leftMap.equals(rightMap);
+    }
+
+    @UserFunction
+    @Description("apoc.diff.properties([leftNodeId],[rightNodeId])")
+    public boolean properties(@Name("leftNodeId") Long leftNodeId, @Name("rightNodeId") Long rightNodeId) {
+        Node leftNode = db.getNodeById(leftNodeId);
+        Node rightNode = db.getNodeById(rightNodeId);
+
+        return !leftNode.getAllProperties().equals(rightNode.getAllProperties());
+    }
+
+    @UserFunction
+    @Description("apoc.diff.relationships([leftRelationshipId],[rightRelationshipId])")
+    public boolean relationships(@Name("leftRelationshipId") Long leftRelationshipId, @Name("rightRelationshipId") Long rightRelationshipId) {
+        Relationship leftRelationship = db.getRelationshipById(leftRelationshipId);
+        Relationship rightRelationship = db.getRelationshipById(rightRelationshipId);
+
+        if ((leftRelationship.getEndNodeId() == rightRelationship.getEndNodeId())
+                && (leftRelationship.getType().name().equals(rightRelationship.getType().name()))
+                && (leftRelationship.getAllProperties().equals(rightRelationship.getAllProperties()))) {
+            return false;
+        }
+        return true;
+    }
+
+    @Procedure(mode = Mode.READ)
+    @Description("apoc.diff.nodes([leftNodeId],[rightNodeId])")
+    public Stream<DiffResult> nodes(@Name("leftNodeId") Long leftNodeId, @Name("rightNodeId") Long rightNodeId) {
+        Node leftNode = db.getNodeById(leftNodeId);
+        Node rightNode = db.getNodeById(rightNodeId);
+
+        DiffResult result = new DiffResult();
+        result.leftOnly = getPropertiesLeftOnly(leftNode, rightNode);
+        result.rightOnly = getPropertiesRightOnly(leftNode, rightNode);
+        result.inCommon = getPropertiesInCommon(leftNode, rightNode);
+        result.differing = getPropertiesDiffering(leftNode, rightNode);
+
+        return Stream.of(result);
+    }
+
+    private Map<String, Object> getPropertiesLeftOnly(Node leftNode, Node rightNode) {
+        Map<String, Object> leftOnly = new HashMap<>();
+        leftOnly.putAll(leftNode.getAllProperties());
+        leftOnly.putAll(rightNode.getAllProperties());
+        leftOnly.keySet().removeAll(rightNode.getAllProperties().keySet());
+        return leftOnly;
+    }
+
+    private Map<String, Object> getPropertiesRightOnly(Node leftNode, Node rightNode) {
```

Comment:
it's the same as `getPropertiesLeftOnly` just reversed parameters. no need for two methods :)